component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -982,6 +1030,175 @@ private boolean isEmailOTPDisableForUser(String username, AuthenticationContext
         return false;
     }
 
+    /**
+     * Email OTP handled for local users.
+     *
+     * @param username name of the user
+     * @param authenticatedUser {@link AuthenticatedUser} object of the authenticated user
+     * @param context {@link AuthenticationContext} object of the authentication request
+     * @param emailOTPParameters {@link Map} of email OTP authenticator specific parameters
+     * @param isEmailOTPMandatory boolean value to identify whether email OTP is configured as mandatory or not
+     * @param queryParams extracted query parameters from the context
+     * @param request {@link HttpServletRequest}
+     * @param response {@link HttpServletResponse}
+     * @throws AuthenticationFailedException when anything failed during handling email OTP for local user.
+     */
+    private void handleEmailOTPForLocalUser(String username, AuthenticatedUser authenticatedUser,
+                                            AuthenticationContext context, Map<String, String> emailOTPParameters,
+                                            boolean isEmailOTPMandatory, String queryParams,
+                                            HttpServletRequest request, HttpServletResponse response)
+            throws AuthenticationFailedException {
+        try {
+            if (isEmailOTPDisableForUser(username, context, emailOTPParameters)) {
+                if (isEmailOTPMandatory) {
```

Comment:
Check whether the "isEmailOTPMandatory" first and give priority to the servers preference

Reply:
fixed